app/assets/javascripts/3dbio_viewer/src/data/repositories/NMRApiRepository.ts

```diff
@@ -0,0 +1,173 @@
+import { nmrFragmentCodec, NMRScreeningFragment } from "../NMRScreening";
+import { getResults, Pagination, paginationCodec } from "../codec-utils";
+import { getValidatedJSON } from "../request-utils";
+import { BasicNMRFragmentTarget, NMRFragmentTarget } from "../../domain/entities/Protein";
+import { Future } from "../../utils/future";
+import i18n from "../../domain/utils/i18n";
+
+export class NMRApiRepository implements NMRRepository {
+    getPartialNMRTarget(
+        target: BasicNMRFragmentTarget,
+        pagination: NMRPagination
+    ): FutureData<{ target: NMRFragmentTarget; pagination: NMRPagination }> {
+        const { uniprotId, start, end } = target;
+        const { bionotesStaging } = routes;
+        const nmrTarget$ = getValidatedJSON<Pagination<NMRScreeningFragment>>(
+            `${bionotesStaging}/bws/api/nmr/${uniprotId}?start=${start}&end=${end}&limit=${
+                pagination.pageSize
+            }&page=${pagination.page + 1}`,
+            paginationCodec(nmrFragmentCodec)
+        ).flatMap(p =>
+            getNMRTarget(uniprotId, getResults(p)).map(target => ({
+                pagination: p
+                    ? { ...pagination, page: pagination.page - 1, count: p.count }
+                    : pagination,
+                target,
+            }))
+        );
+
+        return nmrTarget$;
+    }
+
+    getNMRTarget(target: BasicNMRFragmentTarget): FutureData<NMRFragmentTarget> {
+        const { uniprotId, start, end } = target;
+        const { bionotesStaging } = routes;
+        const chunkSize = 50;
+        const targetChunk$ = (page: number) =>
+            getValidatedJSON<Pagination<NMRScreeningFragment>>(
+                `${bionotesStaging}/bws/api/nmr/${uniprotId}?start=${start}&end=${end}&limit=${chunkSize}&page=${
+                    page + 1
+                }`,
+                paginationCodec(nmrFragmentCodec)
+            ).flatMap(pagination => getNMRTarget(uniprotId, getResults(pagination)));
+
+        const nmrTarget$ = getValidatedJSON<Pagination<NMRScreeningFragment>>(
+            `${bionotesStaging}/bws/api/nmr/${uniprotId}?start=${start}&end=${end}&limit=1`,
+            paginationCodec(nmrFragmentCodec)
+        )
+            .flatMap(pagination => {
+                const pages = Math.ceil((pagination?.count ?? 0) / chunkSize);
+                return Future.sequential(_.times(pages).map(page => targetChunk$(page)));
+            })
+            .map(targets =>
+                targets.reduce((acc, v) => ({
+                    ...acc,
+                    bindingCount: acc.bindingCount + v.bindingCount,
+                    notBindingCount: acc.notBindingCount + v.notBindingCount,
+                    fragments: [...acc.fragments, ...v.fragments],
+                }))
+            );
+
+        return nmrTarget$;
+    }
+
+    saveNMRTarget(target: NMRFragmentTarget) {
+        const targetKeys = ["name", "uniprotId", "start", "end", "fragments"];
```

Comment:
This  `save` does not belong to NMRApiRepository, note how we are not using the api. In the past, we'd create some export repository + a JSON data implementation. However, Jorge's advice is not to have import/export features as repositories, instead do it in the presentation (webapp, script) layer.  So the use case simply gets the entity values (getNMRTarget) and the generation of the JSON+save is done in the react app (to keep it clean, in a different file)